src/components/PopupWithForm.js

```diff
@@ -32,15 +43,26 @@ export default class PopupWithForm extends Popup {
     });
   }
 
+  _renderLoading(isLoading) {
+    if (isLoading) {
+      this._submitButton.textContent = "Saving...";
+    } else {
+      this._submitButton.textContent = this._initialButtonText;
+    }
+  }
+
   setEventListeners() {
     super.setEventListeners();
     this._popupForm.addEventListener("submit", (evt) => {
       evt.preventDefault();
+      this._renderLoading(true);
       this._handleFormSubmit({
         inputData: this._getInputValues(),
         form: this.getForm(),
+      }).finally(() => {
+        this._renderLoading(false);
+        this.close();
```

Comment:
Please, delete it

` this.close()` should be in `then`. You close the modals in `then` in `index.js` correctly

You need to close modals only in `then` blocks after a **successful response** from the server so that when there is an error, it would be opened (the inputs will not be cleared and the user could send the data again to the server), and if you change the button text, you could see the change as well. **Please, correct it everywhere in the code**

Reply:
actually i did that because I thought there is should be like form reset before submit its like if user just closed without submit. not I think I get it.